src/lib.rs

```diff
@@ -1,13 +1,93 @@
-use zonebuilder;
-use pyo3::prelude::*;
+#[allow(unused)]
+use geo::{map_coords::MapCoordsInplace, Point};
+use pyo3::{prelude::*, types::PyDict, Python};
+use pythonize::pythonize;
+use zonebuilder::{self, Params};
 
 #[pyfunction]
 fn triangular_sequence(n: usize) -> Vec<f64> {
     zonebuilder::triangular_sequence(n)
 }
 
+#[pyclass]
+#[derive(Clone)]
+struct Pyparams {
+    num_segments: usize,
+    distances: Vec<f64>,
+    num_vertices_arc: usize,
+    precision: usize,
+    projected: bool,
+}
+impl Default for Pyparams {
+    fn default() -> Self {
+        Pyparams {
+            num_segments: 12,
+            distances: triangular_sequence(5),
+            num_vertices_arc: 10,
+            precision: 6,
+            projected: false,
+        }
+    }
+}
+impl Pyparams {
+    fn to_zb_params(&self) -> Params {
+        Params {
+            num_segments: self.num_segments,
+            distances: self.distances.clone(),
+            num_vertices_arc: self.num_vertices_arc,
+            precision: self.precision,
+            projected: self.projected,
+        }
+    }
+}
+
+#[pyfunction(kwargs = "**")]
+fn kwargsparse(kwargs: Option<&PyDict>) -> PyResult<Pyparams> {
+    let defaults = Pyparams::default();
+    if kwargs.is_none() {
+        return Ok(defaults);
+    }
+    let udict = kwargs.unwrap();
+    let params = Pyparams {
+        num_segments: match udict.get_item("num_segments") {
+            Some(value) => value.extract()?,
+            None => defaults.num_segments,
+        },
+        distances: match udict.get_item("distances") {
+            Some(value) => value.extract()?,
+            None => defaults.distances,
+        },
+        num_vertices_arc: match udict.get_item("num_vertices_arc") {
+            Some(value) => value.extract()?,
+            None => defaults.num_vertices_arc,
+        },
+        precision: match udict.get_item("precision") {
+            Some(value) => value.extract()?,
+            None => defaults.precision,
+        },
+        projected: match udict.get_item("projected") {
+            Some(value) => value.extract()?,
+            None => defaults.projected,
+        },
+    };
+    Ok(params)
+}
+
+#[pyfunction]
+fn clockboard(center: [f64; 2], kwargs: Option<&PyDict>) -> PyResult<PyObject> {
+    let center_point = Point::new(center[0], center[1]);
+    let params = kwargsparse(kwargs)?;
+    let clockboard = zonebuilder::clockboard(center_point, params.to_zb_params());
+    let gil = Python::acquire_gil();
+    let py = gil.python();
+    Ok(pythonize(py, &clockboard)?)
```

Comment:
What's the resulting Python code look like to interact with the result?

If performance isn't of utmost importance, a simpler approach might be to just return the raw JSON string, and do `json.loads` on the other side

Reply:
It would be a dump and a load like this
```python
import geojson
dump = geojson.dumps(json_dict_from_clockboard)
geojson_obj = geojson.loads(dump)
```
This will give a `geojson.feature.FeatureCollection` and I think people can import it into `geopandas` or whatever else from there. Yes, I agree that we should just return a string so that the user can simply load.

Reply:
Ah, so the resulting `geojson::GeoJson` Rust struct somehow gets Pythonized as a valid object for the python `geojson` crate? If so, that could be useful, but... I have my doubts this magically works out? Returning a string through the interface dodges the issue of mapping the Rust structure into something Pythonic

Reply:
Yeah, the magic happens somewhere in the `pythonize` crate. But life is much simpler dealing with the string, changed it now.